code/datums/emotes.dm

```diff
@@ -81,6 +86,31 @@
 	else
 		user.visible_message(msg, visible_message_flags = EMOTE_MESSAGE, emote_prefix = prefix)
 
+	if(icon && icon_state && isliving(user)) //For temporary mob icon changes
+		var/mob/living/L = user
+
+		L.emoting = TRUE //Pause icon tick updates
+		L.icon = icon
+		L.icon_state = icon_state
+
+		addtimer(CALLBACK(src, .proc/emote_icon_deactivate, L), duration)
+		if(disable_control)
+			L.apply_status_effect(STATUS_EFFECT_STUN, duration)
+
+	if(SFX) //For temporary visuals
+		new SFX(user) //put it at the user's location
+
+
+/datum/emote/proc/emote_icon_deactivate(mob/living/L) //Reverting temp icon changes
+
+	if(!L) //sanity
+		return
+
```

Comment:
Either make this a mob proc so the timers on the emoter or make this a QDELETED check